lib/stages/pulsarPostProcess.cpp

```diff
@@ -268,31 +268,32 @@ void pulsarPostProcess::main_thread() {
                              beam_coord, thread_ids);
             }
 
+            // now store samples into output buffer.
+            for (uint32_t thread_id = 0; thread_id < _num_gpus; ++thread_id) {
```

Comment:
I think this for loop should go back to where it was. If it is up here, `fill_headers` will be looping once every `thread_id` (freq), which is not necessary. All the 4 freq goes into one packet.

Reply:
thanks for the clairification @cherryng. i've reverted the for-loop and incrementing calls to their original ordering. now, the only change is the indexing calculation.